appveyor.yml

```diff
@@ -0,0 +1,53 @@
+environment:
+  matrix:
+    # Windows & python 3.6
+    - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
+      PYTHON: "C:\\Python36-x64"
+      PYTHON_ARCH: "64"
+
+    # Windows & python 3.7
+    - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
+      PYTHON: "C:\\Python37-x64"
+      PYTHON_ARCH: "64"
+
+    # Ubuntu & python 3.6
+    - APPVEYOR_BUILD_WORKER_IMAGE: Ubuntu
+      PYTHON: "3.6"
+
+    # Ubuntu & python 3.7
+    - APPVEYOR_BUILD_WORKER_IMAGE: Ubuntu
+      PYTHON: "3.7"
+
+stack: python %PYTHON%
+
+install:
+  - "git submodule update --init --recursive"
+  - cmd: "SET PATH=%PYTHON%;%PATH%"
+  - cmd: "SET PATH=C:\\Python36-x64\\Scripts;%PATH%"
+  - cmd: "SET PATH=C:\\Python37-x64\\Scripts;%PATH%"
+  - "python -V"
+  - "pip --version"
+  - cmd: "python -m  pip install -r requirements.txt"
+  - cmd: "python -m  pip install -r requirements-dev-windows.txt"
+  - sh: "pip install -r requirements.txt"
+  - sh: "pip install -r requirements-dev-linux.txt"
+  - sh: "sudo add-apt-repository -y ppa:deadsnakes/ppa "
+  - sh: "sudo apt-get update"
+  - sh: "sudo apt-get -y install python3.6-dev"
+  - sh: "sudo apt-get -y install python3.7-dev"
+
+
+build: off
+
+test_script:
+  - cmd: "test_windows.bat"
+  - sh: "export QT_QPA_PLATFORM=offscreen"
+  - sh: "bash test_linux.sh"
+
+after_test:
+  - cmd: "7z a zubax-kucher.7z *"
+  - sh: "7z a zubax-kucher.7z * &>/dev/null &"
+  - "appveyor PushArtifact zubax-kucher.7z"
+  - "pyinstaller pyinstaller.spec"
```

Comment:
Invoking PyInstaller directly is not quite correct. On GNU/Linux, we must use StaticX to produce a truly portable executable, as implemented in this build script: https://github.com/Zubax/kucher/blob/appveyor-CI/build_linux.sh. Running just PyInstaller is not sufficient since the resulting executable ends up being [dependent on a particular version of libGL](https://github.com/Zubax/kucher/issues/1), which is disastrous for redistributable executables. So when building for GNU/Linux, invoke the above script. For consistency, I recommend defining an equivalent build script for Windows, despite the fact that no additional steps are needed there (yet).

Now, there was a problem with StaticX that I described here: https://github.com/JonathonReinhart/staticx/issues/79. Jonathon said that I should look into the dependencies to try and find out which one pulls in the unnecessary stuff. Shortly after receiving that reply I realized that the problem might be with the PyQtGraph library, which is no longer needed anyway (see https://github.com/Zubax/kucher/issues/21). I suggest to remove PyQtGraph and then see if StaticX is able to link a portable executable.